### PR TITLE
refactor: improve logging, remove data volume overlap, pass passwords as file

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
- Improve logging and exception handling.
- Remove volumes that are overlapping with data volume  as the postgres db fails to initialize when the data volume is not empty/not a valid data directory. 
- Instead use commands supported by postgres for config file initialization - https://www.postgresql.org/docs/9.3/runtime-config-file-locations.html
- Pass passwords as files instead of env vars
- Remove secrets key check as we no longer manage container based on key path updates. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/constants.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 DB_CREDENTIAL_SECRET_KEY = "DBCredentialSecret"
 POSTGRES_USERNAME_KEY = "POSTGRES_USER"
```
Comment (avipinku) [inline src/constants.py:8]: Do we also want a `POSTGRES_USERNAME_KEY_FILE`?

File: src/container.py [hunk 2]
```diff
@@ -86,37 +93,62 @@ def _recreate_container(self, configuration):
...
+                "Could not remove the container: {}-{}  as it does not exist : {}".format(
+                    self.postgresql_container.name, self.postgresql_container.id, e.explanation
+                )
+            )
 
     def _get_volumes(self, config: ComponentConfiguration):
-        volumes = [f"{config.get_host_volume()}:{DEFAULT_CONTAINER_VOLUME}"]
+        volumes = [
+            f"{config.get_host_volume()}:{DEFAULT_CONTAINER_VOLUME}",
+            f"{PASSWORDS_PATH}:{CUSTOM_FILES}/{POSTGRES_PASSWORD_FILE_KEY}",
```
Comment (avipinku) [inline src/container.py:126]: Nice

File: src/container.py [hunk 3]
```diff
@@ -125,12 +157,28 @@ def _run_container(self, config: ComponentConfiguration):
         )
         self._follow_container_logs()
 
+    def write_secrets_to_file(self, db_password):
+        with open(PASSWORDS_PATH, "w") as p_file:
```
Comment (avipinku) [inline src/container.py:161]: error handling here?

**Review by avipinku — APPROVED** [review]

File: src/container.py [hunk 4]
```diff
@@ -125,12 +156,38 @@ def _run_container(self, config: ComponentConfiguration):
...
+                u_file.write(db_username)
+            with open(db_password_path_, "w") as p_file:
+                p_file.write(db_password)
+        except Exception as e:
+            logging.exception("Exception while writing the secrets files: ", e)
+
+    def _create_config_command(self, config):
+        command = ""
+        server_configuration_files = config.get_pg_config_files()
+        if server_configuration_files:
```
Comment (Nelsonochoam) [inline src/container.py:176]:
```suggestion
        if not server_configuration_files:
            return
            
       for conf_file in server_configuration_files.keys():
                command = command + " -c {}={}".format(SUPPORTED_CONFIGURATION_FILES[conf_file], f"{CUSTOM_FILES}/{conf_file}")
            
       return  command
```

File: src/container.py [hunk 5]
```diff
@@ -125,12 +156,38 @@ def _run_container(self, config: ComponentConfiguration):
...
         self._follow_container_logs()
 
+    def _write_secrets_to_file(self, db_username, db_password):
+        db_user_path = self.secrets_path.joinpath(POSTGRES_USERNAME_FILE_KEY).resolve()
+        db_password_path_ = self.secrets_path.joinpath(POSTGRES_PASSWORD_FILE_KEY).resolve()
+        try:
+            # Create secrets directory if it
+            self.secrets_path.mkdir(parents=True, exist_ok=True)
+            # Write secret files
+            with open(db_user_path, "w") as u_file:
```
Comment (Nelsonochoam) [inline src/container.py:166]: What system user would write this files?  - If we are writing them to the files system could we make it read only to whatever role we are running greengrass as?

Comment (saranyailla) [reply src/container.py:166]: this will be written by greengrass component user. That file is accessed by root and that greengrass component user only.

**Review by Nelsonochoam — APPROVED** [review]

**Review by avipinku — APPROVED** [review]

**Pull request merged**